src/main/java/de/tum/in/ase/notification/CustomFeedbackParser.java

```diff
@@ -1,52 +1,56 @@
-package de.tum.in.www1.jenkins.notifications;
+package de.tum.in.ase.notification;
+
+import com.google.gson.Gson;
+import de.tum.in.ase.notification.model.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.InvalidPropertiesFormatException;
 import java.util.List;
 import java.util.Optional;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-import javax.annotation.Nonnull;
-
-import com.google.gson.Gson;
-
-import de.tum.in.www1.jenkins.notifications.model.*;
-
-import hudson.FilePath;
-import hudson.model.TaskListener;
-
 public class CustomFeedbackParser {
 
+    private static final Logger LOGGER = LogManager.getLogger();
+
     private CustomFeedbackParser() {
     }
 
     /**
      * Reads all {@link CustomFeedback} JSONs from the directory and converts them into a {@link Testsuite}
      *
-     * @param taskListener needed to provide useful error messages in the Jenkins log.
      * @param resultsDir the directory to read the feedback files from.
      * @return a {@link Testsuite} if at least one valid {@link CustomFeedback} has been found, empty otherwise.
-     * @throws IOException if the resultsDir is not readable.
-     * @throws InterruptedException if the resultsDir is not readable.
+     * @throws IOException          if the resultsDir is not readable.
      */
-    public static Optional<Testsuite> extractCustomFeedbacks(@Nonnull TaskListener taskListener, FilePath resultsDir) throws IOException, InterruptedException {
+    public static Optional<Testsuite> extractCustomFeedbacks(Path resultsDir) throws IOException {
+        LOGGER.debug("Extracting custom feedbacks from " + resultsDir);
```

Comment:
```suggestion
        LOGGER.debug("Extracting custom feedbacks from {}", resultsDir);
```
The logger provides its own formatting implementation. That should be used instead to lazily build the strings.

Should be checked for other places that use the logger as well.